zilencer/views.py

```diff
@@ -53,6 +53,19 @@ def remote_server_register_push(request, entity, user_id=REQ(),
 
     return json_success()
 
+@has_request_variables
+def remote_server_register_push_notification(request, hostname, contact_email, uuid, api_key):
```

Comment:
This name is super confusing -- it's hard to tell how it differs from `remote_server_register_push`, and in fact the answer has nothing to do with the difference in the names.

Let's call it `remote_server_register_server`. Also let's put it above `remote_server_register_push` -- that function logically belongs next to its "unregister" friend, and this function is logically first in that it's what you do before you can do any of the others.